echidna/core/smear.py

```diff
@@ -233,6 +237,81 @@ def random_gaussian_energy_spectra(self, true_spectrum):
         smeared_spectrum._raw_events = raw_events
         return smeared_spectrum
 
+
+    def random_gaussian_energy_spectra_parallel(self, true_spectrum,nProcs):
+        """ Smears the energy of a spectra object using multiprocessing  by generating
+          a number of random points from a Gaussian pdf generated
+          for that bin. The number of points generated is equivalent
+          to the number of entries in that bin.
+
+        Args:
+          true_spectrum (spectra): spectrum to be smeared
+
+        Returns:
+          A smeared spectra object.
+        """
+        def Worker(true_spectrum,lower,upper,out_q):
+            smeared_chunk = spectra.Spectra(
+                true_spectrum._name+str(self._light_yield)+"_light_yield",
+                true_spectrum._num_decays)
+            raw_events = true_spectrum._raw_events
+            energy_step = (true_spectrum._energy_high-true_spectrum._energy_low)/true_spectrum._energy_bins
+            time_step = (true_spectrum._time_high-true_spectrum._time_low)/true_spectrum._time_bins
+            radial_step = (true_spectrum._radial_high-true_spectrum._radial_low)/true_spectrum._radial_bins
+            for time_bin in range(int(lower),int(upper)):
+                mean_time = time_bin*time_step+0.5*time_step
+                for radial_bin in range(true_spectrum._radial_bins):
+                    mean_radius = radial_bin*radial_step+0.5*radial_step
+                    for energy_bin in range(true_spectrum._energy_bins):
+                        mean_energy = energy_bin*energy_step+0.5*energy_step
+                        sigma = self.get_energy_sigma(mean_energy)
+                        entries = true_spectrum._data[energy_bin,
+                                                      radial_bin,
+                                                      time_bin]
+                        for i in range(int(entries)):
+                            try:
+                                #print "Filling arry"
+                                num = np.fabs(np.random.normal(mean_energy, sigma))
+                                #print "Fill NUM: ",num
+                                smeared_chunk.fill(num,mean_radius, mean_time)
+                                #print "SUM OF CHUNK: ",smeared_chunk.sum() 
+                            except ValueError:
+                                # Occurs when smeared energy is outside bin range
+                                print "Warning: Smeared energy out of bounds. Skipping."
+                                continue
+       
+            out_q.put(smeared_chunk)
+        
+        smeared_spectrum = spectra.Spectra(
+            true_spectrum._name+str(self._light_yield)+"_light_yield",
+            true_spectrum._num_decays)
+        chunksize =  math.ceil(true_spectrum._time_bins/float(nProcs))
```

Comment:
Is it possible to use `numpy.ceil` instead of `math.ceil` to make use of existing imports, rather than having to import a new module?